google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -471,12 +471,11 @@ def transaction &block
             tx = Transaction.from_grpc(tx_grpc, session)
             begin
               block.call tx
-            rescue Google::Cloud::AbortedError
-              # TODO: retrieve delay from ABORTED error
-              # Retry the entire transaction
-              tx2_grpc = @project.service.begin_transaction session.path
-              tx2 = Transaction.from_grpc(tx2_grpc, session)
-              block.call tx2
+            rescue Google::Cloud::AbortedError => err
+              sleep delay_from_aborted(err)
+              tx_grpc = @project.service.begin_transaction session.path
+              tx = Transaction.from_grpc(tx_grpc, session)
+              retry
```

Comment:
So this will just keep on retrying till the txn succeds or fails with some other error? I can imagine some pathological cases where transaction keeps on aborting many times. Why not have a deadline?